Unrecognized field "kty" (class org.keycloak.jose.jwk.JSONWebKeySet), not marked as ignorable (one known property: "keys"])

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

I have two keycloak instances running the same version, 26.2.4,.  The instances started I believe around 22.x and upgraded through the intermittent versions.

When I try to access `Clients>Client details` for an existing client I get an error and the keys tab does not display.  The server logs show `Invalid jwks representation!` and `Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "kty" (class org.keycloak.jose.jwk.JSONWebKeySet), not marked as ignorable (one known property: "keys"])` .

<img width="348" height="186" alt="Image" src="https://github.com/user-attachments/assets/78c4d790-04e7-46c5-8f67-3648efa802e8" />

[keycloak_keys_error.txt](https://github.com/user-attachments/files/22956182/keycloak_keys_error.txt)

If i try to create a new realm and new client, I can get to the keys tab.  But clicking generate keys gives me a `java.security.KeyStoreException: Cannot overwrite own certificate` as seen in keycloak_create_new_keys_error.txt

<img width="670" height="214" alt="Image" src="https://github.com/user-attachments/assets/1e15ca9b-3da8-4c6d-8e74-a18299d6686a" />

[keycloak_create_new_keys_error.txt](https://github.com/user-attachments/files/22956187/keycloak_create_new_keys_error.txt)

### Version

26.2.4

### Regression

- [ ] The issue is a regression

### Expected behavior

To display the keys tab without error and then allow me to generate keys

### Actual behavior

I get error accessing the keys tab  and am unable to generate keys.

### How to Reproduce?

# For existing Realm and client
- Goto existng Realm and client with JWKS keys 
- Click the Keys tab under Client Details
- Get error

# For new Realm 
- Create Realm
- Create Client
- Enable Client Authorization
- Click the keys tab
- Click generate keys and get and error

### Anything else?

I don't know when the behavior started, I want to say I didn't see it in 25.x.  But the last time i can recall interacting with this tab was probably in 24.x. If no one else is seeing it , i am guessing something happened during an update that i missed but am unsure how to check or resolve that.

--- a/js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties
+++ b/js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties
@@ -2173,6 +2173,7 @@ policyEnforcementModeHelp=The policy enforcement mode dictates how policies are 
 selectAUser=Select a user
 groupCreated=Group created
 generateError=Could not generate a new key pair and certificate {{error}}
+getKeyInfoError=Could not read the certificate information {{error}}
 testClusterSuccess=Successfully verified availability for\: {{successNodes}}
 whoWillAppearLinkTextRoles=Who will appear in this user list?
 attestationPreference.not\ specified=Not specified

--- a/js/apps/admin-ui/src/clients/keys/Keys.tsx
+++ b/js/apps/admin-ui/src/clients/keys/Keys.tsx
@@ -69,7 +69,14 @@ export const Keys = ({
   });
 
   useFetch(
-    () => adminClient.clients.getKeyInfo({ id: clientId, attr }),
+    async () => {
+      try {
+        return await adminClient.clients.getKeyInfo({ id: clientId, attr });
+      } catch (error) {
+        addError("getKeyInfoError", error);
+        return {} as CertificateRepresentation;
+      }
+    },
     (info) => setKeyInfo(info),
     [key],
   );

--- a/services/src/main/java/org/keycloak/services/util/CertificateInfoHelper.java
+++ b/services/src/main/java/org/keycloak/services/util/CertificateInfoHelper.java
@@ -75,6 +75,9 @@ public class CertificateInfoHelper {
 
         try {
             JSONWebKeySet keySet = JsonSerialization.readValue(jwks, JSONWebKeySet.class);
+            if (keySet == null || keySet.getKeys() == null) {
+                throw new IllegalStateException("Certificate not found");
+            }
             JWK publicKeyJwk = JWKSUtils.getKeyForUse(keySet, JWK.Use.SIG);
             if (publicKeyJwk == null) {
                 throw new IllegalStateException("Certificate not found for use sig");

